mithril-client-wasm/src/certificate_verification_cache.rs

```diff
@@ -0,0 +1,441 @@
+            expire_at: DateTime<Utc>,
+        ) {
+            let storage = open_local_storage().unwrap().unwrap();
+            let key = self.cache_key(certificate_hash);
+            let existing_value = Self::parse_cached_certificate(
+                storage.get_item(&key).unwrap().expect("Key not found"),
+            )
+            .unwrap();
+            storage
+                .set_item(
+                    &key,
+                    &serde_json::to_string(&CachedCertificate::new(
+                        &existing_value.previous_hash,
+                        expire_at,
+                    ))
+                    .unwrap(),
+                )
+                .unwrap();
+        }
+
+        /// `Test only` Get the cached value for the given certificate hash
+        pub(super) fn get_cached_value(
+            &self,
+            certificate_hash: &CertificateHash,
+        ) -> Option<CachedCertificate> {
+            let storage = open_local_storage().unwrap().unwrap();
+            storage
+                .get_item(&self.cache_key(certificate_hash))
+                .unwrap()
+                .map(Self::parse_cached_certificate)
+                .transpose()
+                .unwrap()
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashMap;
+    use wasm_bindgen_test::*;
+
+    use super::{test_tools::*, *};
+
+    // Note: as those tests are using local storage, they MUST be run in a browser as node doesn't
+    // have support for local storage.
+    wasm_bindgen_test_configure!(run_in_browser);
+
+    #[wasm_bindgen_test]
+    async fn from_str_iterator() {
+        let cache =
+            LocalStorageCertificateVerifierCache::new("from_str_iterator", TimeDelta::hours(1))
+                .with_items([("first", "one"), ("second", "two")]);
+
+        assert_eq!(
+            HashMap::from_iter([
+                ("first".to_string(), "one".to_string()),
+                ("second".to_string(), "two".to_string())
+            ]),
+            cache.content()
+        );
+    }
+
+    mod store_validated_certificate {
+        use super::*;
+
+        #[wasm_bindgen_test]
+        async fn store_in_empty_cache_add_new_item_that_expire_after_parametrized_delay() {
+            let expiration_delay = TimeDelta::hours(1);
+            let now = Utc::now();
+            let cache = LocalStorageCertificateVerifierCache::new(
+                "store_in_empty_cache_add_new_item_that_expire_after_parametrized_delay",
+                expiration_delay,
+            );
+            cache
+                .store_validated_certificate("hash", "parent")
+                .await
+                .unwrap();
+
+            let cached = cache
+                .get_cached_value("hash")
+                .expect("Cache should have been populated");
+
+            assert_eq!(1, cache.len());
+            assert_eq!("parent", cached.previous_hash);
+            assert!(cached.expire_at - now >= expiration_delay);
```

Comment:
`now` it's a bit confusing. 
We expect that `expire_at - now` is less than `expiration_delay`
`now` it's the time before we add the cache.
Note: If we want to be more precise, it's possible to make 2 assertions comparing with time before assertion and one with time after assertion.

Reply:
Renamed those variables to `start_time`.